src/main/typescript/node_modules/@atomist/rug/operations/Handlers.ts

```diff
@@ -135,6 +134,39 @@ class Plan {
   }
 }
 
+
+type EventPlanMessage = LifecycleMessage | DirectedMessage
+
+type CommandMessage = ResponseMessage
+
+/**
+ * For returning from Event Handlers
+ */
+export class EventPlan extends Plan {
+
+  public add?(msg: EventPlanMessage): this {
+    return super.add(msg);
+  }
+
+  static ofMessage(m: EventPlanMessage): EventPlan {
+    return new EventPlan().add(m);
+  }
+}
+
+/**
+ * Plans returned from Command Handlers
+ */
+export class CommandPlan extends Plan {
+
+  public add?(msg: CommandMessage): this {
+    return super.add(msg);
+  }
+
+  static ofMessage(m: CommandMessage): CommandPlan {
```

Comment:
Interesting, couldn't we use generics in the Plan base class for that method? And not declare that class method here and in EventPlan?